Ruby-lsp is completely ignored with Ruby extension installed

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Install the ruby extension
2. Install the ruby-lsp gem with `gem install ruby-lsp`
3. Configuring settings as follows (from the ruby extension docs):
```
{
  "lsp": {
    "ruby-lsp": {
      "initialization_options": {
        "enabledFeatures": {
          "diagnostics": false
        }
      }
    },
    "languages": {
      "Ruby": {
        "language_servers": ["ruby-lsp", "rubocop", "!solargraph", "..."]
      }
    }
  }
}
```
4. Open a ruby file.

No language server runs, and language hints are not available for the ruby file. There is no error message displayed anywhere, and opening logs with "open language server logs" shows only the option of solargraph and not ruby-lsp, with no output for solargraph.

### Environment

Zed: v0.152.4 (Zed)
OS: Linux Wayland fedora 40
Memory: 31 GiB
Architecture: x86_64
GPU: Intel(R) Graphics (ADL GT2) || Intel open-source Mesa driver || Mesa 24.1.6

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
[Zed.log](https://github.com/user-attachments/files/17022110/Zed.log)

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

I have a similar problem, here is my config:

```json
{
  "languages": {
    "Ruby": {
      "language_servers": ["ruby-lsp", "!solargraph", "!rubocop", "..."],
      "format_on_save": "on"
    }
  },
  "lsp": {
    "ruby-lsp": {
      "initialization_options": {
        "enabledFeatures": {
          "diagnostics": false,
          "formatting": true
        },
        "formatter": "rubocop"
      }
    }
  }
}
```

And Zed has this error:

```
2024-09-17T15:05:15.261006+02:00 [INFO] starting language server "ruby-lsp", path: "/Users/erik/Desktop/Business/Altsson", id: 5
2024-09-17T15:05:15.26177+02:00 [ERROR] failed to start language server "ruby-lsp": ruby-lsp must be installed manually. Install it with `gem install ruby-lsp`.
2024-09-17T15:05:15.262009+02:00 [ERROR] server stderr: Some("")
2024-09-17T15:05:15.262228+02:00 [INFO] retrying installation of language server "ruby-lsp" in 1s
2024-09-17T15:05:16.269493+02:00 [INFO] About to spawn test binary
2024-09-17T15:05:16.270049+02:00 [WARN] test binary failed to launch
2024-09-17T15:05:16.270416+02:00 [WARN] test binary check failed
2024-09-17T15:05:16.27078+02:00 [INFO] beginning to reinstall server
2024-09-17T15:05:16.271191+02:00 [INFO] deleting server container
2024-09-17T15:05:16.272008+02:00 [ERROR] server container removal

Caused by:
    No such file or directory (os error 2)
2024-09-17T15:05:16.272429+02:00 [INFO] starting language server "ruby-lsp", path: "/Users/erik/Desktop/Business/Altsson", id: 6
2024-09-17T15:05:16.273413+02:00 [ERROR] failed to start language server "ruby-lsp": ruby-lsp must be installed manually. Install it with `gem install ruby-lsp`.
2024-09-17T15:05:16.273744+02:00 [ERROR] server stderr: Some("")
2024-09-17T15:05:16.27401+02:00 [ERROR] Hit 4 reinstallation attempts for "ruby-lsp"
```

So it very much looks like it's a PATH issue.. but I can't figure out how to debug this. Clearly it's not finding my "ruby-lsp" command (it works in my terminals).

--- a/docs/src/languages/ruby.md
+++ b/docs/src/languages/ruby.md
@@ -136,12 +136,18 @@ gem install ruby-lsp
 
 Ruby LSP uses pull-based diagnostics which Zed doesn't support yet. We can tell Zed to disable it by adding the following to your `settings.json`:
 
-```json
+```jsonc
 {
+  "languages": {
+    "Ruby": {
+      "language_servers": ["ruby-lsp", "!solargraph", "..."]
+    }
+  },
   "lsp": {
     "ruby-lsp": {
       "initialization_options": {
         "enabledFeatures": {
+          // This disables diagnostics
           "diagnostics": false
         }
       }
@@ -164,10 +170,23 @@ Rubocop has unsafe autocorrection disabled by default. We can tell Zed to enable
 
 ```json
 {
+  "languages": {
+    "Ruby": {
+      // Use ruby-lsp as the primary language server and rubocop as the secondary.
+      "language_servers": ["ruby-lsp", "rubocop", "!solargraph", "..."]
+    }
+  },
   "lsp": {
     "rubocop": {
       "initialization_options": {
         "safeAutocorrect": false
+      }
+    },
+    "ruby-lsp": {
+      "initialization_options": {
+        "enabledFeatures": {
+          "diagnostics": false
+        }
       }
     }
   }

--- a/docs/src/languages/ruby.md
+++ b/docs/src/languages/ruby.md
@@ -140,19 +140,19 @@ Ruby LSP uses pull-based diagnostics which Zed doesn't support yet. We can tell 
 {
   "languages": {
     "Ruby": {
-      "language_servers": ["ruby-lsp", "!solargraph", "..."]
-    }
+      "language_servers": ["ruby-lsp", "!solargraph", "..."],
+    },
   },
   "lsp": {
     "ruby-lsp": {
       "initialization_options": {
         "enabledFeatures": {
           // This disables diagnostics
-          "diagnostics": false
-        }
-      }
-    }
-  }
+          "diagnostics": false,
+        },
+      },
+    },
+  },
 }
 ```
 

--- a/docs/src/languages/ruby.md
+++ b/docs/src/languages/ruby.md
@@ -136,7 +136,7 @@ gem install ruby-lsp
 
 Ruby LSP uses pull-based diagnostics which Zed doesn't support yet. We can tell Zed to disable it by adding the following to your `settings.json`:
 
-```jsonc
+```json
 {
   "languages": {
     "Ruby": {

